lib/liquid_voting_web/schema/custom_types/uuid.ex

```diff
@@ -0,0 +1,36 @@
+defmodule LiquidVotingWeb.Schema.CustomTypes.UUID do
+  @moduledoc """
```

Comment:
Interesting, this is new to me. What are the benefits of using a custom type and checking for the correct uuid format at the graphql layer (as opposed to having Ecto barf when the format is bad)?

Reply:
Both would be fine; to me it's about better modelling the data that's passed in/out of the application.  Internally, when dealing with a `uuid`, it's of type `Ecto.UUID`. Regardless of whether we're passing in a valid `uuid` or not, it should be consistently represented and that implies casting with `Ecto.UUID.cast/1`. Thoughts?

Reply:
Not sure. The issue of representation consistency seems a little abstract to me. Would a string representation for an integer id be cast before letting Ecto handle it?  

In practice does Ecto actually require the UUID string to be cast before finding a record with it?

Meant to confirm this myself, but am amidst `mix deps.get` conflicts here at the moment. https://stackoverflow.com/a/46799252/358518 says `Repo.get!` will try to cast the string itself.

Reply:
In that case, we should probably decode it as a `string` and rely on `Ecto` to do the casting 🤔 
(Here are the absinthe docs on custom scalar types if anyone's curious: https://hexdocs.pm/absinthe/custom-scalars.html)

Reply:
Yeah, I did check those out. I'm not against it, I like having a specific type in terms of semantics. I'd just like to understand the practical gain, if any. With the `datetime` custom scalar example it's not clear to me either.

Also good for us to understand how this will affect consumer clients. What's the behavior if the cast fails on our end? I see the `:error` being returned by `decode`, but what's the final response like? 200 OK with error fields, or a 400 Bad Request?

Reply:
This isn't resolved :) Two questions to be answered here: what is the gain of having the custom type exactly, and how does it change the final response, if at all?

Reply:
I don't feel qualified to comment on this.

Reply:
This could be researched, but if you feel like enough effort has been put into this PR (which is understandable), one option might be to remove this change and open a separate draft PR for further discussion, to get references with time etc. No rush in adding something we don't fully understand at this point.

Reply:
sure. I'll try removing it